src/ast/LlvmVisitor.java

```diff
@@ -109,67 +163,232 @@ public void visit(MainClass mainClass) {
 
     @Override
     public void visit(SysoutStatement sysoutStatement) {
-
+        sysoutStatement.arg().accept(this);
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "\tcall void (i32) @print_int(i32 %%_%d)\n", methodCurrRegIndex-1
+        ));
     }
 
     @Override
     public void visit(AssignStatement assignStatement) {
-
+        String assigneeName = assignStatement.lv();
+        SymbolTable enclosingMethodST = assignStatement.enclosingScope();
+        VariableIntroduction assigneeVarIntro = (VariableIntroduction) enclosingMethodST.getSymbol(assigneeName, false).declaration();
+        String assigneeLLType = getLLVMType(assigneeVarIntro);
+        /**
+         * Case 1: assignee is a local variable in the method
+         */
+        if (enclosingMethodST.contains(assigneeName, false)) {
+            assignStatement.rv().accept(this);
+            LLVMProgram = LLVMProgram.concat(String.format(
+                    "\tstore %s %%_%d, %s* %%%s\n", assigneeLLType, methodCurrRegIndex-1, assigneeLLType, assigneeName
+            ));
+        }
+        String enclosingClassName = enclosingMethodST.getParent().scopeName();
+        List<STSymbol> classInstanceShape = instanceTemplates.get(enclosingClassName);
+        /**
+         * Case 2: assignee is a field of %this
+         */
+        if (classInstanceHasField(classInstanceShape, assigneeName)) {
+            assignStatement.rv().accept(this);
+            int assignedValReg = methodCurrRegIndex - 1;
+            int offset = calcFieldOffset(classInstanceShape, assigneeName);
+            int assigneePtrReg = methodCurrRegIndex;
+            LLVMProgram = LLVMProgram.concat(String.format(
+                    "\t%%_%d = getelementptr i8, i8* %this, i32 %d\n", assigneePtrReg, offset
+            ));
+            methodCurrRegIndex++;
+            int assigneePtrRegPostCast = methodCurrRegIndex;
+            LLVMProgram = LLVMProgram.concat(String.format(
+                    "\t%%_%d = bitcast i8* %%_%d to %s*\n", assigneePtrRegPostCast, assigneePtrReg, assigneeLLType
+            ));
+            methodCurrRegIndex++;
+            LLVMProgram = LLVMProgram.concat(String.format(
+                    "\tstore %s %%_%d, %s* %%_%d\n", assigneeLLType, assignedValReg, assigneeLLType, assigneePtrRegPostCast
+            ));
+        }
+        // else?? do nothing?? there's obvs an error in the code, can it be considered valid?
     }
 
     @Override
     public void visit(AssignArrayStatement assignArrayStatement) {
-
+        String assigneeName = assignArrayStatement.lv();
+        SymbolTable enclosingMethodST = assignArrayStatement.enclosingScope();
+        String enclosingClassName = enclosingMethodST.getParent().scopeName();
+        List<STSymbol> classInstanceShape = instanceTemplates.get(enclosingClassName);
+
+        VariableIntroduction assigneeVarIntro = (VariableIntroduction) enclosingMethodST.getSymbol(assigneeName, false).declaration();
```

Comment:
use STLookup.findDeclTable() instead of getSymbol to avoid null pointer exception